pgml-extension/pgml_extension/model.py

```diff
@@ -916,10 +935,43 @@ def fit(self, snapshot: Snapshot):
     def deploy(self, strategy):
         """Promote this model to the active version for the project that will be used for predictions"""
 
-        plpy.execute(
-            f"""
+        deployment_id = dict(
+            plpy.execute(
+                f"""
             INSERT INTO pgml.deployments (project_id, model_id, strategy, created_at) 
             VALUES ({q(self.project_id)}, {q(self.id)}, {q(strategy)}, clock_timestamp())
+            RETURNING id
+        """
+            )[0]
+        )["id"]
+
+        plpy.execute(
+            f"""
+        CREATE OR REPLACE FUNCTION pgml.predict(
+            project_name TEXT,          -- Human-friendly project name
+            features DOUBLE PRECISION[] -- Must match the training data column order
+        )
+        RETURNS DOUBLE PRECISION
+        AS $$
+            from pgml_extension.model import Project
+
+            return float(Project.find_by_name(project_name, {q(deployment_id)}).deployed_model.predict(features))
```

Comment:
Isn't deployment id model-specific? In this case, we are treating the `pgml.predict` function as model-specific but it's not.

Reply:
The original implementation uncached all projects, when any model was deployed so it was overly aggressive. Now we'll check if the project's model id has changed after a deploy, and only reload the model if it is no longer the currently deployed one.